### PR TITLE
Adding Tool Error tracking

### PR DESCRIPTION
**Pull request by kellrott**
Traditionally errors related tool xml formatting or issues that occurred during processing the command line (the 'failure preparing job' message) have required tool developers to dig into server logs to find out what is causing the problem. 
This PR adds a system to track those errors and view them via the API and a page on the admin panel.

### CODE REVIEW

File: lib/galaxy/tools/__init__.py [hunk 1]
```diff
@@ -80,6 +82,24 @@
...
+class ToolErrorLog:
+    def __init__(self):
+        self.error_stack = []
+        self.max_errors = 100
+    
+    def add_error(self, file, section, exception):
+        self.error_stack.insert(0, {
+            "file" : file,
+            "time" : str(datetime.now()),
+            "section" : section,
```
Comment (nsoranzo) [inline lib/galaxy/tools/__init__.py]: What about "stage" or "phase" instead of "section"?

File: templates/admin/tool_errors.mako [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+<h2>Tool Errors</h2>
+
+<p>
+Internal Tool Error log
+</p>
+
+<table class="manage-table colored" border="0" cellspacing="0" cellpadding="0" width="100%">
+<tr class="header">
+<td>Time</td>
+<td>System</td>
```
Comment (nsoranzo) [inline templates/admin/tool_errors.mako]: As above, what about "stage" or "phase"?

**Comment by kellrott** [comment]
I changed the wording to use 'Phase' to describe when an error was occurring. 
Also I removed the API component, based on offline comments from @jmchilton

**Comment by nsoranzo** [comment]
:+1:

**Comment by dannon** [comment]
+1

**Pull request merged**